integration_test/third_party_apps_test.go

```diff
@@ -407,16 +407,40 @@ func verifyLog(actualLog *cloudlogging.Entry, expectedLog *metadata.ExpectedLog)
 	return nil
 }
 
+// stripUnavailableFields removes the fields that are listed as unavailable_on
+// the current platform.
+func stripUnavailableFields(fields []*metadata.LogFields, platform string) []*metadata.LogFields {
+	var result []*metadata.LogFields
+	for _, field := range fields {
+		if !metadata.SliceContains(field.UnavailableOn, platform) {
+			result = append(result, field)
+		}
+	}
+	return result
+}
+
 func runLoggingTestCases(ctx context.Context, logger *logging.DirectoryLogger, vm *gce.VM, logs []*metadata.ExpectedLog) error {
```

Comment:
Since this is related, can we remove the hack related to b/239240173 on line 602 (bug is marked as fixed)?
If we can not, change it to this new format.

Reply:
I updated the bug number. I think the bug is still not fixed. I'm not sure what you mean about changing it to the new format; the feature in this PR cannot be used for aerospike + SLES because this PR only includes a feature for missing log FIELDS, not logs that are entirely missing. It wouldn't be hard to add that feature though, do you want me to do that in this PR or a separate one?

Reply:
ah ok, got it. Thanks for updating it.